src/PowerShellEditorServices/Session/PowerShell3Operations.cs

```diff
@@ -32,27 +33,11 @@ public void PauseDebugger(Runspace runspace)
             out DebuggerResumeAction? debuggerResumeAction)
         {
             IEnumerable<TResult> executionResult = null;
-
-            using (var nestedPipeline = currentRunspace.CreateNestedPipeline())
+            using (var pwsh = PowerShell.Create())
             {
-                foreach (var command in psCommand.Commands)
-                {
-                    nestedPipeline.Commands.Add(command);
-                }
-
-                var results = nestedPipeline.Invoke();
-
-                if (typeof(TResult) != typeof(PSObject))
-                {
-                    executionResult =
-                        results
-                            .Select(pso => pso.BaseObject)
-                            .Cast<TResult>();
-                }
-                else
-                {
-                    executionResult = results.Cast<TResult>();
-                }
+                pwsh.Runspace = currentRunspace;
```

Comment:
I seem to recall seeing somewhere that setting `Runspace` here to `currentRunspace` does something special. It might be worth a comment about that if so.

Reply:
What special does it do? :)

Reply:
Like it replaces nested pipelines or something? Maybe I'm remembering wrong...

Reply:
I believe your thinking of `PowerShell.Create(RunspaceMode.CurrentRunspace)`.  In this case it's just a field, nothing special.

Reply:
Ok perfect!